ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -181,14 +212,18 @@
        (format "Subscription with with provider-id [%s] and native-id [%s] already exists."
                provider-id
                native-id)))
-    (let [new-subscription (api-core/body->concept!
+    (let [tmp-subscription (api-core/body->concept!
                             :subscription
                             provider-id
                             native-id
                             body
                             content-type
                             headers)
-          subscriber-id (get-subscriber-id new-subscription)]
+          subscriber-id (get-subscriber-id tmp-subscription)
+          token-user (api-core/get-user-id-from-token request-context)
+          new-subscription (-> tmp-subscription
+                               (add-subscriber-id-if-missing request-context token-user)
```

Comment:
I don't see the need to create two functions, one to add subscriber-id and one to add email. You ended up parsing the subscription twice and having to create a function that takes in a function. It would be easier and simpler to just add a function that add in subscriber-id and email -- call it something like `add-subscriber-and-email-if-missing`.

Reply:
Code has been simplified using this idea - 👍